Documentation/dev-guide/api_grpc_gateway.md

```diff
@@ -0,0 +1,38 @@
+
+## Why grpc-gateway
+
+etcd v3 uses [gRPC][grpc]. Most recommended is to use etcd [Go client][go-client] and [etcdctl][etcdctl]. However, some languages are not supported by gRPC. This is where [grpc-gateway][grpc-gateway] steps in. It builds a RESTful proxy based on the gRPC services, which can serve HTTP and JSON requests.
```

Comment:
etcd v3 uses [gRPC][grpc] for its messaging protocol. The etcd project includes a gRPC-based [Go client][go-client] and a command line utility, [etcdctl][etcdctl], for communicating with an cluster etcd through gRPC. For languages with no gRPC support, etcd provides a JSON [grpc-gateway][grpc-gateway]. This gateway serves a RESTful proxy that translates HTTP/JSON requests into gRPC messages.